src/pages/strategy/graphable.js

```diff
@@ -1,165 +1,149 @@
-						}
-					}
-				} else {
-					for(j=0;j<(Object.keys(itemData.name).length);j++) {
-						// Check if its not the first x-axis
-						if(graphIndex > 0) {
-							// Use the values from previous record
-								data[itemData.name[j]][graphIndex] = data[itemData.name[j]][graphIndex-1];
-						} else {
-							// First x-axis with no record, use zero :: if there's no data beyond the axis
-								data[itemData.name[j]][graphIndex] = (nearest!==undefined) ? this[dataKey][nearest][itemData.dbkey[j]] : 0;
-						}
-					}
-				}
-				graphIndex++;
+		this.collectData({
+			tableName: this.tableName,
+			graphableItems: this.graphableItems,
+			start: $("#startDate").val(),
+			end: $("#endDate").val(),
+			interval: $("#graphInterval").val(),
+			delta: $("#deltaMode").prop("checked")
+		});
+	}
+	
+	// Get data via worker using specified filters
+	collectData(filters){
+		let DataCollector = new Worker(chrome.extension.getURL('library/workers/graph-data.js'));
+		DataCollector.onmessage = (response) => {
+			if (response.data) {
+				this.refreshGraph(response.data);
+			} else {
+				this.loadingGraph = false;
+				$(".graph_title").text("Unable to render");
+				$(".graph_input").prop("disabled", false);
 			}
-			
-			// Draw graph
-			this.ctx[i] = $("#chart" + (i+1)).get(0).getContext("2d");
-			this.chart[i] = new Chart( this.ctx[i] ).Line(this.buildGraphData(data), {});
+			DataCollector.terminate();
 		};
+		DataCollector.postMessage(Object.assign({
+			url: chrome.extension.getURL(""),
+			playerId: PlayerManager.hq.id
+		}, filters));
+	}
+	
+	// Re-draw the graph and date title
+	refreshGraph(data){
+		// Show graph title dates
+		$(".graph_title").text(
+			new Date($("#startDate").val()).format("mmm d, yyyy")
+			+" to "
+			+new Date($("#endDate").val()+" 00:00:00").format("mmm d, yyyy")
+		);
 		
-		/* Util: Build charting parameters
-		--------------------------------------------*/
-		this.dataset = function(label, color1, color2, dataList){
-			return {
-				label: label,
-				fillColor: "rgba(225,225,225,0)",
-				strokeColor: "rgba("+color1+",1)",
-				pointColor: "rgba("+color1+",1)",
-				pointStrokeColor: "#"+color2,
-				pointHighlightFill: "#"+color2,
-				pointHighlightStroke: "rgba("+color1+",1)",
-				data: dataList
-			};
-		};
+		// Ability to hide types
+		this.graphableItems.dbkey.forEach(function(dbkey, ind){
+			if (!$(".legend_toggle input[data-type=\""+dbkey+"\"]").prop("checked")) {
+				delete data.datasets[dbkey];
+			}
+		});
 		
-		this.buildGraphData = function(data){
-			var self = this;
-			return {
-				labels: data.hours,
-				datasets: Array.apply(null,{length:Object.keys(itemData.name).length}).map(function(c,i){
-					return self.dataset(itemData.full[i], itemData.colorbyte[i], itemData.colorhex[i], data[itemData.name[i]]);
-				}).reverse(),
-			};
-		};
+		// New chart JS only accepts an array, not object
+		data.datasets = Object.values(data.datasets);
```

Comment:
NOTE: `Object.values` is ES7 feature, which supported from chrome 54.
see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/values